web/modules/custom/joinup_front_page/joinup_front_page.behat.inc

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/**
+ * @file
+ * Contains \JoinupFrontPageSubContext.
+ */
+
+declare(strict_types = 1);
+
+use Behat\Gherkin\Node\TableNode;
+use Drupal\DrupalExtension\Context\DrupalSubContextBase;
+use Drupal\DrupalExtension\Context\DrupalSubContextInterface;
+use Drupal\joinup\Traits\EntityTrait;
+
+/**
+ * Behat step definitions for testing invitations.
+ */
+class JoinupFrontPageSubContext extends DrupalSubContextBase implements DrupalSubContextInterface {
+
+  use EntityTrait;
+
+  /**
+   * Loads and pins entities in the front page.
+   *
+   * @param string $type
+   *   The entity type id.
+   * @param \Behat\Gherkin\Node\TableNode $table
+   *   A list of titles.
+   *
+   * @throws \InvalidArgumentException
+   *   Thrown when the user has entered an invalid type.
+   *
+   * @Given site pinned :type entities:
```

Comment:
I would not use "site pinned" as wording because this is not how it is understood by the end user. How about: 

```
@Given the following :type entities are pinned to the front page:
```